### PR TITLE
Fix-ups prior to "Module Resolution Strategies"

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Some fix-ups to #181. @mhegazy

### CODE REVIEW

File: pages/Module Resolution.md [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
- If none were found, you will get `error TS2307: Cannot find module 'moduleA'.`
+Finally, if the compiler could not resolve the module, it will log an error.
+In this case, the error would be something like `error TS2307: Cannot find module 'moduleA'.`
 
 ## Relative vs. Non-relative module imports
 
 Module imports are resolved differently based on whether the module reference is relative or non-relative.
 
-A **relative** import is one that stars with `/`, `./` or `../`.
+A *relative import* is one that stars with `/`, `./` or `../`.
```
Comment (sandersn) [inline pages/Module Resolution.md]: starts

File: pages/Module Resolution.md [hunk 2]
```diff
@@ -1,42 +1,46 @@
 > This section assumes some basic knowledge about modules.
 Please see the [Modules](./Modules.md) documentation for more information.
 
-"Module Resolution" refers to the process the compiler follows to understand the shape of an import target.
-Consider an import statement like `import { a } from "./moduleA"`; to be able to correctly check any use of the import target `a`, the compiler needs to know what it means, and what is its type.
-To do that, the compiler needs to inspect the exported declaration with name `a` from `moduleA`.
-This could be defined in a `.ts` file from your own code, or in a `.d.ts` for an external definition file that your code depends on.
-The first step, is to locate the definition of `moduleA`.
+*Module resolution* is the process the compiler takes to figure out exactly what an import refers to.
```
Comment (sandersn) [inline pages/Module Resolution.md]: I would substitute "uses" for "takes" (at least I think this is the usual verb) and "decide" for "figure out exactly" (much less wordy).

File: pages/Module Resolution.md [hunk 3]
```diff
@@ -1,42 +1,46 @@
...
+Consider an import statement like `import { a } from "moduleA"`;
+in order to check any use of `a`, the compiler needs to know exactly what it represents, and will need to check its definition `moduleA`.
 
-The process to locate a module goes as follows:
+At this point, the compiler will ask itself "what's the shape of `moduleA`?"
+While this sounds straightforward, `moduleA` could be defined in one of your own a `.ts`/`.tsx` files, or in a `.d.ts` that your code depends on.
```
Comment (sandersn) [inline pages/Module Resolution.md]: Extra "a" before .ts

File: pages/Module Resolution.md [hunk 4]
```diff
@@ -1,42 +1,46 @@
...
 
-* Locate a file with the module definition
+First, the compiler will try to locate a file that represents the imported module.
+To do so the compiler follows one of two different strategies: [Classic](#classic) or [Node](#node).
+These strategies tell the compiler *where* to look for `moduleA`.
 
- The compiler will try to first locate a file containing the definition of the imported module.
- To do so the compiler follows one of two different strategies: [Classic](#classic) and [Node](#node).
+If that didn't work and if the module name is non-relative (and in the case of `"moduleA`, it is), then the compiler will attempt to locate an [ambient module declaration](#ambient-module-declarations).
```
Comment (sandersn) [inline pages/Module Resolution.md]: Missing " after "moduleA"

File: pages/Module Resolution.md [hunk 5]
```diff
@@ -1,42 +1,46 @@
 > This section assumes some basic knowledge about modules.
 Please see the [Modules](./Modules.md) documentation for more information.
 
-"Module Resolution" refers to the process the compiler follows to understand the shape of an import target.
```
Comment (sandersn) [inline pages/Module Resolution.md]: I'm not sold on the rewrite of this first section. The original is pretty good and just needs some grammar fixes and maybe light conversion to a prose style.

File: pages/Module Resolution.md [hunk 6]
```diff
@@ -1,42 +1,46 @@
...
 
 Module imports are resolved differently based on whether the module reference is relative or non-relative.
 
-A **relative** import is one that stars with `/`, `./` or `../`.
+A *relative import* is one that stars with `/`, `./` or `../`.
+Some examples include:
+
+    * `import Entry from "./components/Entry";`
+    * `import { DefaultHeaders } from "../constants/http";`
+    * `import "/mod";`
```
Comment (sandersn) [inline pages/Module Resolution.md]: 'Relative' is a weird name for an absolute path. Just saying.

File: pages/Module Resolution.md [hunk 7]
```diff
@@ -1,42 +1,46 @@
...
-Consider an import statement like `import { a } from "./moduleA"`; to be able to correctly check any use of the import target `a`, the compiler needs to know what it means, and what is its type.
-To do that, the compiler needs to inspect the exported declaration with name `a` from `moduleA`.
-This could be defined in a `.ts` file from your own code, or in a `.d.ts` for an external definition file that your code depends on.
-The first step, is to locate the definition of `moduleA`.
+*Module resolution* is the process the compiler takes to figure out exactly what an import refers to.
+Consider an import statement like `import { a } from "moduleA"`;
+in order to check any use of `a`, the compiler needs to know exactly what it represents, and will need to check its definition `moduleA`.
 
-The process to locate a module goes as follows:
+At this point, the compiler will ask itself "what's the shape of `moduleA`?"
```
Comment (mhegazy) [inline pages/Module Resolution.md]: do not think you need "itself" here, it is implied.

File: pages/Module Resolution.md (hunk 6 above)
Comment (mhegazy) [inline pages/Module Resolution.md]: i do not know what else to call ti to be frank. rooted?

**Comment by mhegazy** [comment]
No more comments for me. off to @sandersn

**Comment by mhegazy** [comment]
I will hold off on comments in #181 to make the merge easier.

File: pages/Module Resolution.md (hunk 6 above)
Comment (sandersn) [inline pages/Module Resolution.md]: I don't have any ideas either, just complaining. :)

Comment (DanielRosenwasser) [inline pages/Module Resolution.md]: From http://www.ecma-international.org/ecma-262/6.0/#sec-hostresolveimportedmodule:

> A typical normalization process would include actions such as alphabetic case folding and expansion of relative and abbreviated path specifiers.

So it looks like the important distinction is *abbreviated* and *non-abbreviated* paths, and abbreviated paths are what we consider "non-relative" here. But maybe @rbuckton and @bterlson might know more about this.

Comment (DanielRosenwasser) [inline pages/Module Resolution.md]: Though, "abbreviated" might be vague enough that we shouldn't use it. I'm inclined to omit rooted/absolute paths.

**Pull request merged**